File: Socket_program/serverudp.py
#!/usr/bin/env python3
import socket

def Main():
    host = '127.0.0.1'
    port = 50001

    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    s.bind((host, port))
    print("Server Started")

    while True:
        data, addr = s.recvfrom(1024)
        message = data.decode('utf-8')
        print("Message from: " + str(addr))
        print("From connected user: " + message)

        # Echo the message back to the client
        s.sendto(data, addr)

if __name__ == '__main__':
    Main()

File: Socket_program/serverudp.c
#include <stdio.h>
#include <arpa/inet.h>
#include <fcntl.h>
#include <unistd.h>
#include <sys/socket.h>

int main()
{
    int welcome; //new_soc, fd, n;
    char buffer[1024], msg[100];
    struct  sockaddr_in addr; //

    welcome = socket(PF_INET, SOCK_DGRAM, 0);

    addr.sin_family = AF_INET;
    addr.sin_port = htons(7891);
    addr.sin_addr.s_addr = inet_addr("127.0.0.1");

    bind(welcome, (struct sockaddr *) &addr, sizeof(addr));
    printf("\nServer is Online");
    recvfrom(welcome, msg,sizeof(msg), 0,(struct sockaddr*)&addr, sizeof(addr));
    printf("Message: %s",msg);
	    
//	    sendto(welcome,msg, sizeof(msg),0,(struct sockaddr*)&addr, sizeof(addr));
   
   
   
    close(welcome);

    return 0;
}
